### PR TITLE
Remove docker api dependency from cli/config

### PR DESCRIPTION
**Pull request by tiborvass**
Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>
Signed-off-by: Tibor Vass <tibor@docker.com>

### CODE REVIEW

File: cli/config/credentials/file_store.go [hunk 1]
```diff
@@ -62,3 +63,18 @@ func (c *fileStore) GetFilename() string {
 func (c *fileStore) IsFileStore() bool {
 	return true
 }
+
+// ConvertToHostname converts a registry url which has http|https prepended
+// to just an hostname.
+func ConvertToHostname(url string) string {
```
[the remark this answers is not in the document]
Comment (tiborvass) [reply cli/config/credentials/file_store.go]: @thaJeztah I doubt. We could maybe add a test that imports it and makes sure it matches, but the new `go mod` stuff that I'm playing with downloads test dependencies too. May be overkill. If things break we'll update. Alternatively, we could add a nice comment to say "please keep in sync with X from other repo". WDYT @tonistiigi ?

File: cli/config/configfile/file.go [hunk 2]
```diff
@@ -195,7 +194,7 @@ func (configFile *ConfigFile) Save() error {
 
 // ParseProxyConfig computes proxy configuration by retrieving the config for the provided host and
 // then checking this against any environment variables provided to the container
-func (configFile *ConfigFile) ParseProxyConfig(host string, runOpts []string) map[string]*string {
+func (configFile *ConfigFile) ParseProxyConfig(host string, runOpts map[string]*string) map[string]*string {
```
[the remark this answers is not in the document]
Comment (tiborvass) [reply cli/config/configfile/file.go]: I don't think they are equivalent refactors. I'm trying to remove dependency of docker/docker from cli/config. It's not pretty but this is what I have :)

File: cli/config/credentials/file_store.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (tonistiigi) [reply cli/config/credentials/file_store.go]: The only place it is used seems to be where legacy format is used that is super old and should be removed. The whole registry pkg is pretty much unused and last parts of it should probably go away with containerd.

**Comment by tonistiigi** [comment]
@thaJeztah mergeable?

File: cli/command/container/run.go [hunk 3]
```diff
@@ -68,7 +68,7 @@ func NewRunCommand(dockerCli command.Cli) *cobra.Command {
 }
 
 func runRun(dockerCli command.Cli, flags *pflag.FlagSet, ropts *runOptions, copts *containerOptions) error {
-	proxyConfig := dockerCli.ConfigFile().ParseProxyConfig(dockerCli.Client().DaemonHost(), copts.env.GetAll())
+	proxyConfig := dockerCli.ConfigFile().ParseProxyConfig(dockerCli.Client().DaemonHost(), opts.ConvertKVStringsToMapWithNil(copts.env.GetAll()))
```
Comment (thaJeztah) [inline cli/command/container/run.go]: https://github.com/docker/cli/pull/1573 was merged; can you rebase, and update `runCreate()` as well? https://github.com/docker/cli/blob/91bc4ddde205fca60c9932322b34c4808fd0733c/cli/command/container/create.go#L64-L66

File: cli/config/credentials/file_store.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (thaJeztah) [reply cli/config/credentials/file_store.go]: Can you add a comment referring to the other implementation to keep them in sync?

**Pull request merged**